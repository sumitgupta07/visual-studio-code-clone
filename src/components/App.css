.header {
  min-width: 100vh;
  margin-bottom: 1em;
  margin-top: 0;
  font-size: 1.5em;
}
.content {
  text-align: left;
  display: flex;
  flex-direction: row;
  align-items: stretch;
  justify-content: stretch;
  color: #333;
  background: #eee;
  min-height: 100vh;
  margin: 2em;
}
.sidebar {
  flex: 2;
  border: 1px solid #2291b1;
  display: flex;
  flex-direction: column;
}

.navbar {
  flex: 10;
  border: 1px solid #2291b1;
  padding: 5px;
}
.editorpane {
  flex: 24;
  border: 1px solid #2291b1;
}
li {
  list-style-type: none;
}
.file-list {
  border-bottom: 1px solid #ccc;
  padding-left: 0;
}
.file-list-item {
  display: inline-block;
  list-style: none;
  margin-bottom: -1px;
  padding: 0.5rem 0.75rem;
}
.file-active {
  background-color: white;
  border: solid #ccc;
  border-width: 1px 1px 0 1px;
}
