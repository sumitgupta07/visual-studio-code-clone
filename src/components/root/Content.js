import React, { useState } from 'react';
import NavBar from '../pane/NavBar';
import SelectPane from '../pane/SelectPane';
import EditorPane from '../pane/EditorPane';
import { folders } from '../utils/constants';

const Content = () => {
  //Initialize selected Nav Item
  const [selectedNav, setSelectedNav] = useState('File Explorer');

  //Initialize selected file with first file in first folder
  const [selectedFile, setSelectedFile] = useState(folders[0].files[0]);

  const handleSelectNav = selectedItem => {
    setSelectedNav(selectedItem);
  };

  const handleSelectFile = file => {
    setSelectedFile(file);
  };

  return (
    <div className="content">
      <NavBar onSelectNav={handleSelectNav} />
      <SelectPane
        currentSelection={selectedNav}
        onSelectFile={handleSelectFile}
      />
      <EditorPane activeFile={selectedFile} onSelectFile={handleSelectFile} />
    </div>
  );
};

export default Content;
