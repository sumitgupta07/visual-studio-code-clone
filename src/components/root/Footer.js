import React from 'react';

const Footer = () => {
  return <div>Contact Us</div>;
};

export default Footer;
