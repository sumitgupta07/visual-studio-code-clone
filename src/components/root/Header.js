import React from 'react';

const Header = () => {
  return <div className="header">Visual Studio Code</div>;
};

export default Header;
