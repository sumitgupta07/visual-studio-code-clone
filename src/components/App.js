import React, { Fragment } from 'react';
import './App.css';
import Header from './root/Header';
import Content from './root/Content';
import Footer from './root/Footer';

const App = () => (
  <Fragment>
    <Header />
    <Content />
    <Footer />
  </Fragment>
);
export default App;
