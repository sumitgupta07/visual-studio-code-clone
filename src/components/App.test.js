import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import App from './App';
import Header from './root/Header';
import Content from './root/Content';
import Footer from './root/Footer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render App component', () => {
  const wrapper = mount(<App />);
  expect(wrapper).toMatchSnapshot();
});

it('renders Header component', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Header).length).toEqual(1);
});

it('renders Content component', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Content).length).toEqual(1);
});

it('renders Footer component', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Footer).length).toEqual(1);
});
