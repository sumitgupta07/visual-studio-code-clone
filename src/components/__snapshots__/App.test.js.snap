// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
<App>
  <Header>
    <div
      className="header"
    >
      Visual Studio Code
    </div>
  </Header>
  <Content>
    <div
      className="content"
    >
      <NavBar
        onSelectNav={[Function]}
      >
        <div
          className="sidebar"
        >
          <Icon
            iconClass="far fa-copy fa-3x"
            id="FileExplorer"
            label="File Explorer"
            onSelectIcon={[Function]}
          >
            <button
              onClick={[Function]}
            >
              <div
                className="tooltip"
              >
                <div
                  className="icon"
                >
                  <i
                    className="far fa-copy fa-3x"
                  />
                </div>
                <div
                  className="tooltiptext"
                >
                  <span>
                    File Explorer
                  </span>
                </div>
              </div>
            </button>
          </Icon>
          <Icon
            iconClass="fas fa-search fa-rotate fa-3x"
            id="Search"
            label="Search"
            onSelectIcon={[Function]}
          >
            <button
              onClick={[Function]}
            >
              <div
                className="tooltip"
              >
                <div
                  className="icon"
                >
                  <i
                    className="fas fa-search fa-rotate fa-3x"
                  />
                </div>
                <div
                  className="tooltiptext"
                >
                  <span>
                    Search
                  </span>
                </div>
              </div>
            </button>
          </Icon>
          <Icon
            iconClass="fas fa-code-branch fa-3x"
            id="SourceControl"
            label="Source Control"
            onSelectIcon={[Function]}
          >
            <button
              onClick={[Function]}
            >
              <div
                className="tooltip"
              >
                <div
                  className="icon"
                >
                  <i
                    className="fas fa-code-branch fa-3x"
                  />
                </div>
                <div
                  className="tooltiptext"
                >
                  <span>
                    Source Control
                  </span>
                </div>
              </div>
            </button>
          </Icon>
          <Icon
            iconClass="fas fa-bug fa-3x"
            id="Debug"
            label="Debug"
            onSelectIcon={[Function]}
          >
            <button
              onClick={[Function]}
            >
              <div
                className="tooltip"
              >
                <div
                  className="icon"
                >
                  <i
                    className="fas fa-bug fa-3x"
                  />
                </div>
                <div
                  className="tooltiptext"
                >
                  <span>
                    Debug
                  </span>
                </div>
              </div>
            </button>
          </Icon>
          <Icon
            iconClass="fas fa-th-large fa-3x"
            id="Extensions"
            label="Extensions"
            onSelectIcon={[Function]}
          >
            <button
              onClick={[Function]}
            >
              <div
                className="tooltip"
              >
                <div
                  className="icon"
                >
                  <i
                    className="fas fa-th-large fa-3x"
                  />
                </div>
                <div
                  className="tooltiptext"
                >
                  <span>
                    Extensions
                  </span>
                </div>
              </div>
            </button>
          </Icon>
        </div>
      </NavBar>
      <SelectPane
        currentSelection="File Explorer"
        onSelectFile={[Function]}
      >
        <FileExplorer
          onSelectFile={[Function]}
        >
          <div
            className="navbar"
          >
            <h3>
              File Explorer
            </h3>
            <i
              className="fas fa-folder-open"
            >
                
              <span>
                src
              </span>
              <ul>
                <li
                  id="a"
                  key="a"
                  onClick={[Function]}
                >
                  <i
                    className="far fa-file"
                  >
                     
                    index.js
                  </i>
                </li>
                <li
                  id="b"
                  key="b"
                  onClick={[Function]}
                >
                  <i
                    className="far fa-file"
                  >
                     
                    App.js
                  </i>
                </li>
                <li
                  id="c"
                  key="c"
                  onClick={[Function]}
                >
                  <i
                    className="far fa-file"
                  >
                     
                    Content.js
                  </i>
                </li>
              </ul>
            </i>
          </div>
        </FileExplorer>
      </SelectPane>
      <EditorPane
        activeFile={
          Object {
            "content": "Placeholder for index.js. It has App.js as child components",
            "fileName": "index.js",
            "id": "a",
          }
        }
        onSelectFile={[Function]}
      >
        <div
          className="editorpane"
        >
          <h3>
            Editor Pane
          </h3>
          <div
            className="files"
          >
            <ul
              className="file-list"
            >
              <FileTab
                activeFile={
                  Object {
                    "content": "Placeholder for index.js. It has App.js as child components",
                    "fileName": "index.js",
                    "id": "a",
                  }
                }
                file={
                  Object {
                    "content": "Placeholder for index.js. It has App.js as child components",
                    "fileName": "index.js",
                    "id": "a",
                  }
                }
                key="a"
                onClick={[Function]}
              >
                <li
                  className="file-list-item file-active"
                  id="a"
                  onClick={[Function]}
                >
                  index.js
                </li>
              </FileTab>
              <FileTab
                activeFile={
                  Object {
                    "content": "Placeholder for index.js. It has App.js as child components",
                    "fileName": "index.js",
                    "id": "a",
                  }
                }
                file={
                  Object {
                    "content": "Placeholder for App.js. It has Header and Content as child components",
                    "fileName": "App.js",
                    "id": "b",
                  }
                }
                key="b"
                onClick={[Function]}
              >
                <li
                  className="file-list-item"
                  id="b"
                  onClick={[Function]}
                >
                  App.js
                </li>
              </FileTab>
              <FileTab
                activeFile={
                  Object {
                    "content": "Placeholder for index.js. It has App.js as child components",
                    "fileName": "index.js",
                    "id": "a",
                  }
                }
                file={
                  Object {
                    "content": "Placeholder for Content.js file. It has Navbar, SelectPane and EditorPane as child components",
                    "fileName": "Content.js",
                    "id": "c",
                  }
                }
                key="c"
                onClick={[Function]}
              >
                <li
                  className="file-list-item"
                  id="c"
                  onClick={[Function]}
                >
                  Content.js
                </li>
              </FileTab>
            </ul>
          </div>
          <FileContent
            currentFile={
              Object {
                "content": "Placeholder for index.js. It has App.js as child components",
                "fileName": "index.js",
                "id": "a",
              }
            }
            key="a"
          >
            <div
              id="a"
            >
              Placeholder for index.js. It has App.js as child components
            </div>
          </FileContent>
        </div>
      </EditorPane>
    </div>
  </Content>
  <Footer>
    <div>
      Contact Us
    </div>
  </Footer>
</App>
`;
